lib/rack/handler/david.rb

```diff
@@ -4,15 +4,22 @@ class David
       def self.run(app, options={})
         g = Celluloid::Supervision::Container.run!
 
-        g.supervise(as: :server, type: ::David::Server, args: [app, options])
-        g.supervise(as: :gc, type: ::David::GarbageCollector)
+        g.supervise(as: :server_udp, type: ::David::Server::CoAP, args: [app, options])
+        if options[:DTLS] == 'true'
+          g.supervise(as: :server_dtls, type: ::David::Server::CoAPs, args: [app, options])
+        end
 
+        g.supervise(as: :gc, type: ::David::GarbageCollector)
         if options[:Observe] != 'false'
           g.supervise(as: :observe, type: ::David::Observe)
         end
 
         begin
-          Celluloid::Actor[:server].run
+          if options[:DTLS] == 'true'
+            Celluloid::Actor[:server_dtls].async.run
```

Comment:
The `David::Server#run` method seems to block and since we want to start two servers here the dtls server is started as `async.run` not sure if that's really a good idea…